extensions-contrib/redis-cache/src/main/java/io/druid/client/cache/RedisCache.java

```diff
@@ -0,0 +1,190 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.client.cache;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.Lists;
+import com.metamx.emitter.service.ServiceEmitter;
+import com.metamx.emitter.service.ServiceMetricEvent;
+import io.druid.java.util.common.logger.Logger;
+import redis.clients.jedis.Jedis;
+import redis.clients.jedis.JedisPool;
+import redis.clients.jedis.JedisPoolConfig;
+import redis.clients.jedis.exceptions.JedisException;
+
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.AtomicReference;
+
+public class RedisCache implements Cache
+{
+    private static final Logger log = new Logger(MemcachedCache.class);
+
+    private JedisPool pool;
+    private RedisCacheConfig config;
+
+    private final AtomicLong hitCount = new AtomicLong(0);
+    private final AtomicLong missCount = new AtomicLong(0);
+    private final AtomicLong timeoutCount = new AtomicLong(0);
+    private final AtomicLong errorCount = new AtomicLong(0);
+
+    private final AtomicReference<AtomicLong> priorRequestCount = new AtomicReference<>(new AtomicLong(0L));
```

Comment:
For thread safe, it should be AtomicReference. Same usage can be seen in CaffeineCache and MemcachedCache. But because of your reminder, I found it can set to AtomicReference<Long>. AtomicReference<AtomicLong> is not necessary. By the way, I have  filled out the Druid Contributor License Agreement.